RUFAS/biophysical/manure/manure_manager.py

```diff
@@ -0,0 +1,455 @@
+        -------
+        None
+        """
+        row_names: list[str] = self._generate_adjacency_matrix_keys()
+
+        for origin_name, connections in processor_connections_by_name.items():
+            is_separator: bool = (origin_name in self._all_separators.keys())
+            if is_separator:
+                self._create_column_in_adjacency_matrix(origin_name, row_names, is_separator)
+                self._populate_destination_proportions(
+                    connections["solid_output_destinations"], f"{origin_name}_solid_output")
+                self._populate_destination_proportions(
+                    connections["liquid_output_destinations"], f"{origin_name}_liquid_output")
+            else:
+                self._create_column_in_adjacency_matrix(origin_name, row_names, is_separator)
+                self._populate_destination_proportions(connections["destinations"], origin_name)
+
+    def _create_column_in_adjacency_matrix(self, origin_name: str, row_names: list[str], is_separator: bool) -> None:
+        """
+        Add a column to the adjacency matrix for a given origin node.
+
+        This method modifies the adjacency matrix to include connections
+        from the specified origin node to a list of destination nodes.
+        For separators, it creates multiple columns representing distinct
+        output types (input, solid output, liquid output). For non-separators,
+        a single column is created.
+
+        Parameters
+        ----------
+        origin_name : str
+            The name of the origin node for which the column(s) will be created.
+        row_names : list[str]
+            The list of destination node names to initialize in the adjacency matrix.
+        is_separator : bool
+            A flag indicating whether the origin node is a separator
+
+        Returns
+        -------
+        None
+        """
+        if is_separator:
+            self._adjacency_matrix[f"{origin_name}_input"] = {
+                destination_name: 0.0 for destination_name in row_names
+            }
+            self._adjacency_matrix[f"{origin_name}_solid_output"] = {
+                destination_name: 0.0 for destination_name in row_names
+            }
+            self._adjacency_matrix[f"{origin_name}_liquid_output"] = {
+                destination_name: 0.0 for destination_name in row_names
+            }
+        else:
+            self._adjacency_matrix[origin_name] = {
+                destination_name: 0.0 for destination_name in row_names
+            }
+
+    def _populate_destination_proportions(self, connections: list[dict[str, Any]], origin_name: str) -> None:
+        """
+        Populate the destination proportions for the given origin in the adjacency matrix.
+
+        This method updates the adjacency matrix to store the proportion of connections from the specified origin
+        to each destination. If the receiving processor name corresponds to an internal separator, its name is
```

Comment:
```suggestion
        to each destination. If the receiving processor name corresponds to an internal separator, its name is
```
I think I'm a little confused on what is meant by "internal separator", can you explain?

Reply:
I think what I meant here by "internal" is referring to checking against "self._all_separators", which all all the created separators

Reply:
So does internal mean a separator that has a config and has been instantiated and added to the matrix?

Reply:
So they have been instantiated and added to `self.all_processors`. But the actual instance is not in the matrix because we changed the key to string

Reply:
What do you think about removing the "internal" description? I don't think I would have thought about what this means if it just said "separator".

Reply:
I agree with Niko's suggestions.